runtime/service.go

```diff
@@ -1088,6 +1152,20 @@ func (s *service) State(requestCtx context.Context, req *taskAPI.StateRequest) (
 		return nil, err
 	}
 
+	// If the fields aren't "", they are pointing in-VM FIFO files.
+	// Replacing them with the corresponding FIFO files on the host.
```

Comment:
Is there ever a case where the fields are empty but we still want to return something here?

Reply:
Let me add test cases around the check. IIRC, /dev/null is converted to "".

Reply:
`cio.NullIO` sets them `""`, but in that case, s.fifos will be also `""`. So I don't need the ifs.

Reply:
I'm going to update this comment.